### PR TITLE
Fix liveness and readiness health endpoint access

### PR DESCRIPTION
**Pull request by PierreBesson**
- Fix an issue where liveness and readiness endpoints were not public which failed Kubernetes deployments.
- Improve `curl` tests to check more sample endpoints

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/src/main/resources/config/application.yml.ejs [hunk 1]
```diff
@@ -125,6 +125,13 @@ management:
         git:
             mode: full
     health:
+        group:
+            liveness:
+                include: livenessState
+            readiness:
+                include: readinessState<% if (databaseType === 'sql'>,datasource<% } %>
```
Comment (pascalgrimaud) [inline generators/server/templates/src/main/resources/config/application.yml.ejs:132]:
```suggestion
                include: readinessState<% if (databaseType === 'sql') { %>,datasource<% } %>
```

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Review by DanielFran — APPROVED** [review]
LGTM

File: generators/server/templates/src/main/java/package/config/ReactiveSecurityConfiguration.java.ejs [hunk 2]
```diff
@@ -266,6 +266,7 @@ public class SecurityConfiguration {
             .pathMatchers("/services/**", "/swagger-resources/**").authenticated()
             <%_ } _%>
             .pathMatchers("/management/health").permitAll()
+            .antMatchers("/management/health/**").permitAll()
```
Comment (murdos) [inline generators/server/templates/src/main/java/package/config/ReactiveSecurityConfiguration.java.ejs:269]: Are you sure it should not be `pathMatchers` here?

File: generators/server/templates/src/main/resources/config/application.yml.ejs [hunk 3]
```diff
@@ -125,6 +125,13 @@ management:
         git:
             mode: full
     health:
+        group:
+            liveness:
+                include: livenessState
+            readiness:
+                include: readinessState<% if (databaseType === 'sql') { %>,datasource<% } %>
+        probes:
+            enabled: true
```
Comment (murdos) [inline generators/server/templates/src/main/resources/config/application.yml.ejs:134]: Do we need when developing locally too, or can it be disabled?

File: generators/server/templates/src/main/java/package/config/ReactiveSecurityConfiguration.java.ejs (hunk 2 above)
Comment (PierreBesson) [reply generators/server/templates/src/main/java/package/config/ReactiveSecurityConfiguration.java.ejs:269]: Oh yes ! Good catch !

File: generators/server/templates/src/main/resources/config/application.yml.ejs (hunk 3 above)
Comment (PierreBesson) [reply generators/server/templates/src/main/resources/config/application.yml.ejs:134]: I think it doesn't impact startup time significantly... So let's have it all the time.

File: generators/server/templates/src/main/java/package/config/ReactiveSecurityConfiguration.java.ejs (hunk 2 above)
Comment (PierreBesson) [reply generators/server/templates/src/main/java/package/config/ReactiveSecurityConfiguration.java.ejs:269]: What is even the difference ? Shouldn't we use antMatcher in all cases ?

Comment (PierreBesson) [reply generators/server/templates/src/main/java/package/config/ReactiveSecurityConfiguration.java.ejs:269]: Oh it's for reactive. @pascalgrimmaud I think the reactive build is not enabled so this error was not caught !

**Comment by PierreBesson** [comment]
Please don't merge until the problem with pathMatcher for reactive build is fixed and the reactive build is enabled.

**Comment by pascalgrimaud** [comment]
Ok @PierreBesson : I convert to draft, so we can't merge it by mistake :)

**Comment by PierreBesson** [comment]
@pascalgrimaud I don't really understand why my builds don't pass :cry:.

**Comment by DanielFran** [comment]
@PierreBesson webflux is broken and there is PR https://github.com/jhipster/generator-jhipster/pull/12688 to fix them

**Comment by mshima** [comment]
Close/reopen to test against current main.

**Comment by mshima** [comment]
Is this PR related to https://github.com/jhipster/generator-jhipster/issues/10916?

**Comment by DanielFran** [comment]
@mshima this might indeed fix the issue with the new probes implemented in SB 2.3, but needs confirmation...

**Review by DanielFran — APPROVED** [review]
LGTM

**Pull request merged**